code/src/java/plugin/pretokens/test/PreArmorTypeTester.java

```diff
@@ -83,13 +84,13 @@ public int passes(final Prerequisite prereq, final CharacterDisplay display, CDO
 				}
 				else
 				{ //not a TYPE string
-					final String eqName = eq.getName().toUpperCase();
+					final String eqName = eq.getName().toUpperCase(Locale.ENGLISH);
```

Comment:
This is certainly data controlled

Reply:
Are data types type case sensitive? Even though this is data controlled, it is an essentially a variable name and not something that should follow the default locale.

Reply:
Not entirely.  The else on line 99/100 is definitely an internal usage that is okay to use Locale.English.  The if on line 93/94 and the else if on line 107/108 are not, and would impact data not written in English.  So in effect, this needs to be properly rewritten to handle that mix of locality (yes, the code is not great today, but to avoid putting other locales in a bad position in the future, let's fix it here)